nonprimary_foreignkey/tests/tests.py

```diff
@@ -0,0 +1,179 @@
+from django.db.models import Prefetch
+from django.test.testcases import TestCase
+from mock import patch
+
+from nonprimary_foreignkey.fields import NonPrimaryForeignKey
+from nonprimary_foreignkey.tests.models import Item
+from nonprimary_foreignkey.tests.models import ItemType
+from nonprimary_foreignkey.tests.models import ReceivedItem
+
+
+class TestGet(TestCase):
+    def setUp(self):
+        Item.objects.all().delete()
+        self.barcode = '12345'
+        self.item = Item.objects.create(barcode=self.barcode)
+        self.barcode2 = '67890'
+        self.item2 = Item.objects.create(barcode=self.barcode2)
+
+    def test_get(self):
+        from_instance = ReceivedItem.objects.create()
+        from_instance.barcode = self.barcode
+        self.assertEqual(from_instance.item, self.item)
+
+    def test_get_non_existent_barcode(self):
```

Comment:
One issue with this behavior is in cases where the item would only _sometimes_ be expected to line up. For example, imagine a situation where all items in inventory are barcoded, but only some are for sale. Additional metadata for sellable inventory is stored in another table. Raising an exception increases the complexity of interacting with the API, though does match the current behavior of a foreign key without a database constraint.